aict_tools/io.py

```diff
@@ -184,24 +343,98 @@ def pickle_model(classifier, feature_names, model_path, label_text='label'):
         joblib.dump(classifier, model_path, compress=4)
 
 
-def append_to_h5py(f, array, group, key):
+class HDFColumnAppender():
     '''
-    Write numpy array to h5py hdf5 file
+    This is a ContextManager which can append columns to an existing hdf5 table 
+    in a chunkwise manner.
+    For hdf5 files in *tables format* this will temprarily occupy twice the disk space.
+    
+    Parameters
+    ----------
+    path: str
+        path to the hdf5 file
+    table_name: str
+        name of the table columns should be appended to
     '''
-    group = f.require_group(group)  # create if not exists
+    def __init__(self, path, table_name):
+        self.path = path
+        self.table_name = table_name
+        try:
+            with pd.HDFStore(path, mode='r') as r:
+                _ = r[table_name]
+            self.is_tables_format = True
+        except TypeError:
+            self.is_tables_format = False
+
+    def __enter__(self):
+        return self
+    
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        if self.is_tables_format:
+            with tables.open_file(self.path, 'r+') as t:
+                try:
+                    t.remove_node(f'/{self.table_name}', recursive='force')
+                    t.rename_node(f'/{self.table_name}_copy', newname=self.table_name)
+                except tables.exceptions.NoSuchNodeError:
+                    pass
+
+    def add_data(self, data, new_column_name, start, stop):
+        '''
+        Appends a column containing new data to existing table.
+
+        Parameters
+        ----------
+        data: array-like
+            the data to append
+        new_column_name: str
+            name of the new column to append
+        start: int or None
+            first row to replace in the file
+        stop: int or None
+            last event to replace in the file
+        '''
+        if self.is_tables_format:
+            with pd.HDFStore(self.path, 'r+') as store:
+                df = store.select(self.table_name, start=start, stop=stop)
+                df[new_column_name] = data
+                # store.remove(self.table_name, start=0, stop=stop-start)
+                store.put(self.table_name + '_copy', df, format='t', append=True)
```

Comment:
Is there any reason to add new keys here? I was having issues writing results to the DataFrame because of the '_copy' part and it seems to work just fine using `store.put(self.table_name, df, format='t')`.

Reply:
In that case the `exit()` funktion wouldnt be needed anymore aswell